Floofbot/Modules/TagCommands.cs

```diff
@@ -33,13 +34,75 @@ public TagCommands(FloofDataContext floofDb)
             _floofDb = floofDb;
         }
 
+        [Command("requireadmin")]
+        [Summary("Set a server to require admin roles for adding or removing tags")]
+        [RequireUserPermission(GuildPermission.Administrator)]
+        public async Task RequireAdmin([Summary("True/False")] string requireAdmin = "")
+        {
+            TagConfig config = await _floofDb.TagConfigs.AsQueryable()
+                .Where(config => config.ServerId == Context.Guild.Id).FirstOrDefaultAsync();
+
+            bool parsedRequireAdmin;
+            if (!bool.TryParse(requireAdmin, out parsedRequireAdmin))
+            {
+                await SendEmbed(CreateDescriptionEmbed($"Usage: `.tag requireadmin True/False`"));
+                return;
+            }
+
+            try
+            {
+                if (config == null)
+                {
+                    await _floofDb.TagConfigs.AddAsync(new TagConfig
+                    {
+                        ServerId = Context.Guild.Id,
+                        TagUpdateRequiresAdmin = parsedRequireAdmin,
+                    });
+                }
+                else
+                {
+                    config.TagUpdateRequiresAdmin = parsedRequireAdmin;
+                }
+                await _floofDb.SaveChangesAsync();
+                string message = $"Adding/removing tags now{(parsedRequireAdmin ? " requires " : " does not require ")}admin permission.";
+                await Context.Channel.SendMessageAsync(message);
+            }
+            catch (DbUpdateException e)
+            {
+                string message = $"Unable to make adding/removing tags{(parsedRequireAdmin ? " " : " not ")}require admin permission.";
+                await Context.Channel.SendMessageAsync(message);
+                Log.Error(e.ToString());
```

Comment:
I think it would be better to give more context to these exceptions eg "Error when configuring tag admin permissions" or something